src/main/java/org/broadinstitute/hellbender/utils/Utils.java

```diff
@@ -1137,4 +1141,128 @@ public static void forceJVMLocaleToUSEnglish() {
         final List<T> sorted = values.stream().sorted().collect(Collectors.toList());
         return sorted.get(sorted.size() / 2);
     }
+
+    public static class LineIterator implements AutoCloseable, Iterable<String> {
+        private final Stream<String> lines;
+
+        /**
+         * Returns an iterator so you can iterate over the lines in the text file like so:
+         * for (String line: new Utils.LineIterator(path)) {
+         *   // do something with the line
+         * }
+         *
+         * It's also closeable so you can close it when done, or use it in a try-with-resources
+         * to close it automatically.
+         *
+         * @param path path to a text file.
+         * @throws UserException if we cannot open the file for reading.
+         */
+        public LineIterator(Path path) throws UserException {
+            try {
+                lines = Files.lines(path);
+            } catch (IOException x) {
+                throw new UserException("Error reading " + path.toString() + ": " + x.getMessage());
```

Comment:
Pass in the nested exception object itself to the `UserException` constructor, not just its message.

Reply:
Done